### ISSUE TITLE
This model performs badly in Chinese and English.

### ISSUE THREAD

**Issue by naginoa** [FrancisGregoire/parSentExtract#8]
Hello，I reproduce this paper with the dataset in WMT. Then I got:
  Training: Loss = 0.000887, Accuracy = 0.9981, Precision = 0.9998, Recall = 0.9889, F1 = 0.9943
  Validing:  Loss = 0.003075, Accuracy = 0.9995, Precision = 0.8858, Recall = 0.6050, F1 = 0.7190
This is very close to the indicator you have in the paper.But, I train and valid this model on en-zh dataset also from WMT, I got:
  Training: Loss = 0.004289, Accuracy = 0.9893, Precision = 0.9993, Recall = 0.9362, F1 = 0.9667
  Validing:  Loss = 0.006478, Accuracy = 0.9995, Precision = 0.4500, Recall = 0.0087, F1 = 0.0171
It is not good as well as en-fr. I tested many other English-Chinese data sets and the same poor performance.
Can you help me？

**Comment by FrancisGregoire** [comment]
Unfortunately I never made experiments with en-zh pair. What values did you use for `decision_threshold`, `num_negative`, `num_epochs`? How many parallel sentence pairs (en-zh) do you have in your train set?

I know that tokenizing corpora and creating vocabularies using sub-word unit approaches such as BPE (see [sentencepiece](https://github.com/google/sentencepiece)) instead of using words as tokens will increase the performance dramatically, but I doubt that it applies to Chinese text.

**Comment by naginoa** [comment]
decision_threshold is 0.99, num_negative is 5, num_epochs is 15.
I trained 500000 parallel sentence pairs (en-zh) .

about the vocabularies, I find your code use space characters to segment words.

tokens = line.strip().split()            line 128 in utils.py

 It is effective for en and fr，but not to zh. So with the help of Senior， we create the vocab using the other chinese word segmentation module. Then I got the result above.
